src/append_only_data.rs

```diff
@@ -205,6 +205,42 @@ impl Permissions for UnpubPermissions {
     fn owner_entry_index(&self) -> u64 {
         self.owner_entry_index
     }
+
+    fn check_permissions_for_key(&self, requester: PublicKey, request: &Request) -> Result<()> {
+        match request {
+            Request::GetAData(..)
+            | Request::GetADataShell { .. }
+            | Request::GetADataRange { .. }
+            | Request::GetADataIndices(..)
+            | Request::GetADataLastEntry(..)
+            | Request::GetADataPermissions { .. }
+            | Request::GetUnpubADataUserPermissions { .. }
+            | Request::GetADataOwners { .. } => {
+                if self.is_action_allowed(requester, Action::Read) {
+                    Ok(())
+                } else {
+                    Err(Error::AccessDenied)
+                }
+            }
+            Request::AddUnpubADataPermissions { .. } | Request::SetADataOwner { .. } => {
```

Comment:
> Request::SetADataOwner

This should be always forbidden for anyone but an owner.
Could you pls move it to the `Request::DeleteAData { .. } => Err(Error::AccessDenied),` block?